code/game/machinery/suit_modifier.dm

```diff
@@ -225,6 +229,32 @@
 	use_power = 1
 	activated = FALSE
 
+/obj/machinery/suit_modifier/proc/process_module_removal(var/mob/living/carbon/human/H)
+	if(activated)
+		return
+	lock_atom(H)
+	activated = TRUE
+	use_power = 2
+	activation_animation()
+	var/obj/item/clothing/suit/space/rig/R = H.is_wearing_item(/obj/item/clothing/suit/space/rig, slot_wear_suit)
+	R.deactivate_suit()
+	if(R.modules.len)
+		// if we have something that's not a rig_module here we have a problem
+		var/obj/item/rig_module/RM = input(H, "Choose an upgrade to remove from [R].", R) as null|anything in R.modules
+		if(!RM|| !H.Adjacent(src) || H.incapacitated())
+			return
```

Comment:
you need to `unlock_atom(H)` before you return or the dude will be stuck in there

Reply:
whoops